src/colvarmodule.cpp

```diff
@@ -386,10 +388,18 @@ int colvarmodule::parse_global_params(std::string const &conf)
   parse->get_keyval(conf, "scriptingAfterBiases",
                     scripting_after_biases, scripting_after_biases);
 
+#if defined(COLVARS_TCL)
+  std::string source_Tcl_script;
+  if (parse->get_keyval(conf, "sourceTclFile", source_Tcl_script)) {
```

Comment:
Since you're getting closer to finishing this, my concern here is that this would force the execution of arbitrary Tcl code in the middle of the Colvars initialization.

I would *at the very least* defer that execution until Colvars is properly set up, and ideally control when that code is executed from the simulation engine.

If you have a Tcl interpreter driving the whole thing as in VMD or NAMD to a partial extent, it would be simpler to just use `source` IMO.

Reply:
> Since you're getting closer to finishing this, my concern here is that this would force the execution of arbitrary Tcl code in the middle of the Colvars initialization.
> 
> I would at the very least defer that execution until Colvars is properly set up, and ideally control when that code is executed from the simulation engine.

Good point. This needs deferring the check for the existence of the force callback until everything is initialized, which I'm doing now.
 
> If you have a Tcl interpreter driving the whole thing as in VMD or NAMD to a partial extent, it would be simpler to just use source IMO.

I agree. This is definitely reserved for the other engines that are not Tcl-driven at all, or to write a Colvars input that is fully portable between one of these two codes and one of the other engines.

Reply:
The `colvarmodule` object itself is constructed separately from the configuration parsing. We just don't use this separation much yet, although I would do that for #418.

So you can test for the presence of the callback functionality after construction, but *before* the configuration is parsed.

However, pretty much everyone will use their Tcl code to modify existing variables or biases, so those would need to be defined before you source the script.